// Copyright 2022 ByteDance and/or its affiliates.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax="proto2";

package monolith;

message MultiHashTableProto {
  repeated int32 dims = 1;
  optional bytes  slot_expire_time_config = 2;
  repeated string table_names = 3;
  optional string learning_rate_tensor = 4;
  optional string shared_name = 5;
  optional int32 saver_parallel = 6;
  optional string initializer_op = 7;
  optional string handle_tensor = 8;
}
